src/Language/Rust.ts

```diff
@@ -0,0 +1,306 @@
+        return keywords;
+    }
+
+    protected forbiddenForClassProperties(_c: ClassType, _className: Name): ForbiddenWordsInfo {
+        return { names: [], includeGlobalForbidden: true };
+    }
+
+    protected forbiddenForUnionMembers(_u: UnionType, _unionName: Name): ForbiddenWordsInfo {
+        return { names: [], includeGlobalForbidden: true };
+    }
+
+    protected forbiddenForEnumCases(_e: EnumType, _enumName: Name): ForbiddenWordsInfo {
+        return { names: [], includeGlobalForbidden: true };
+    }
+
+    protected topLevelNameStyle(rawName: string): string {
+        return camelCaseStyle(rawName);
+    }
+
+    private nullableRustType = (t: Type, withIssues: boolean): Sourcelike => {
+        return ["Option<", this.breakCycle(t, withIssues), ">"];
+    };
+
+    protected isImplicitCycleBreaker(t: Type): boolean {
+        const kind = t.kind;
+        return kind === "array" || kind === "map";
+    }
+
+    private rustType = (t: Type, withIssues: boolean = false): Sourcelike => {
+        return matchType<Sourcelike>(
+            t,
+            _anyType => maybeAnnotated(withIssues, anyTypeIssueAnnotation, "serde_json::Value"),
+            _nullType => maybeAnnotated(withIssues, nullTypeIssueAnnotation, "Option<serde_json::Value>"),
+            _boolType => "bool",
+            _integerType => "i64",
+            _doubleType => "f64",
+            _stringType => "String",
+            arrayType => ["Vec<", this.rustType(arrayType.items, withIssues), ">"],
+            classType => this.nameForNamedType(classType),
+            mapType => ["HashMap<String, ", this.rustType(mapType.values, withIssues), ">"],
+            enumType => this.nameForNamedType(enumType),
+            unionType => {
+                const nullable = nullableFromUnion(unionType);
+
+                if (nullable) return this.nullableRustType(nullable, withIssues);
+
+                const [hasNull,] = removeNullFromUnion(unionType);
+
+                const isCycleBreaker = this.isCycleBreakerType(unionType);
+
+                const name = isCycleBreaker ? ["Box<", this.nameForNamedType(unionType), ">"] : this.nameForNamedType(unionType);
+
+                return hasNull
+                    ? ["Option<", name, ">"] as Sourcelike
+                    : name;
+            }
+        );
+    };
+
+    private breakCycle = (t: Type, withIssues: boolean): any => {
+        const rustType = this.rustType(t, withIssues);
+        const isCycleBreaker = this.isCycleBreakerType(t);
+
+        return isCycleBreaker ? ["Box<", rustType, ">"] : rustType;
+    }
+
+    private emitStructDefinition = (c: ClassType, className: Name): void => {
+        this.emitLine("#[derive(Serialize, Deserialize)]");
+
+        const structBody = () => this.forEachClassProperty(c, "none", (name, jsonName, prop) => {
+
+            const escapedName = rustStringEscape(jsonName);
+
+            this.emitLine("#[serde(rename = \"", escapedName, "\")]");

```

Comment:
I would omit this if `escapedname === jsonName`.

Reply:
The name of the Rust property is not `jsonName`, so that wouldn't be correct.

Reply:
Sorry, yes, I meant if the escapedName matches the property name.

Reply:
We don't do that in most other languages, either.  Like C#.

Reply:
What I'm trying to say is: get the code into a good condition, merge, and only then address cosmetics like this.

Reply:
Sounds good. I do have some trouble reading the Rust output because there are no empty lines between annotations and the preceding properties, although I’m not sure what the Rust convention is.